dask_searchcv/tests/test_adaptive.py

```diff
@@ -0,0 +1,179 @@
+from dask_searchcv.adaptive import _top_k, Hyperband
+from dask_ml.linear_model import PartialSGDClassifier
+from dask_ml.datasets import make_classification
+import numpy as np
+import dask.array as da
+from dask.distributed import Client
+import pandas as pd
+import pytest
+import distributed
+from pprint import pprint
+import scipy.stats as stats
+import random
+from sklearn.linear_model import Lasso
+
+
+class TestFunction:
+    def _fn(self):
+        return self.value
+
+    def get_params(self, deep=None, **kwargs):
+        return {k: getattr(self, k) for k, v in kwargs.items()}
+
+    def set_params(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        return self
+
+    def partial_fit(self, *args, **kwargs):
+        pass
+
+    def score(self, *args, **kwargs):
+        return self._fn()
+
+
+def test_hyperband_sklearn():
```

Comment:
http://distributed.readthedocs.io/en/latest/develop.html#writing-tests may be helpful here.

Async / distributed tests are tricky... I've found the second version, `with cluster()`, to be the easiest to work with.

Reply:
That would remove your `_get_client()` call.

Reply:
I'm still running into testing issues, even after trying `with cluster()` and `gen_cluster(client=True)`. I run into issues even mirroring [distributed/tests/test_client.py#L4717][1]. I *think* I've resolved all the systems level issues, and will check next week.

Any tips? I've pushed some intermingled `gen_cluster` and `with cluster()` code in 22c86e48cf67c36bd88ef3b4f1d9996cd9cadccd.

[1]:https://github.com/dask/distributed/blob/12ddc080d1d876b0fce6fe4b2e863fe7c7b31543/distributed/tests/test_client.py#L4717

Reply:
Could you try this out locally to? I've also had to resolve some systems level issues.